### PR TITLE
Add premerge CI action

### LINKED ISSUE #12: [CICD] Add premerge CI using github action
As container build does not require GPU resource, we could leverage github action resource to implement the premerge CI here

could refer to https://github.com/marketplace/actions/build-and-push-docker-images

and mark **push: false**
```
        name: Build
        uses: docker/build-push-action@v3
        with:
          push: false
          tags: <some PR info>
```

and also https://docs.github.com/en/actions/publishing-packages/publishing-docker-images

### PR DESCRIPTION
**Pull request by YanxuanLiu**
fixed #12 

add pre-merge action to test docker image building.

### CODE REVIEW

**Comment by YanxuanLiu** [comment]
Could you help to check if the tag good? @pxLi

File: .github/workflows/premerge.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# limitations under the License.
+
+name: pre-merge
+
+on:
+  # quick tests for pull requests and the releasing branches
+  push:
+    branches:
+      - dev
+  pull_request:
```
Comment (pxLi) [inline .github/workflows/premerge.yml:18]: specify events like
```
  pull_request:
    types: [opened, synchronize, reopened]
```

File: .github/workflows/premerge.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
```
Comment (pxLi) [inline .github/workflows/premerge.yml:24]: concurrency control
like 
```
concurrency:
  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
  cancel-in-progress: true
```

File: .github/workflows/premerge.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: pre-merge
+
+on:
+  # quick tests for pull requests and the releasing branches
+  push:
+    branches:
```
Comment (pxLi) [inline .github/workflows/premerge.yml]: if we decide to post merge check here, then we should rename the CI to different name.

maybe `container-build-check`

File: .github/workflows/premerge.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      - dev
+  pull_request:
+
+jobs:
+  docker:
+    runs-on: ubuntu-20.04
+    steps:
+      -
```
Comment (pxLi) [inline .github/workflows/premerge.yml]: nit: no need to put a empty line here. could put `- name:`

File: .github/workflows/premerge.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        name: Checkout
+        uses: actions/checkout@v3
+      -
+        name: Build
+        uses: docker/build-push-action@v3
+        with:
+          push: false
+          context: ./Databricks
+          tags: pre-merge-${{ github.sha }}
```
Comment (pxLi) [inline .github/workflows/premerge.yml]: different naming if the CI is not for premerge only

File: .github/workflows/premerge.yml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Build docker image
+        uses: docker/build-push-action@v3
+        with:
+          push: false
+          context: ./Databricks
+          tags: ci-test-${{ github.sha }}
```
Comment (zhanga5) [inline .github/workflows/premerge.yml]: `nit`: no new line at end of file

**Review by zhanga5 — DISMISSED** [review]
LGTM (w/ one minor comment)

**Review by pxLi — APPROVED** [review]

**Pull request merged**